core/app/models/spree/stock/availability_validator.rb

```diff
@@ -5,19 +5,20 @@ def validate(line_item)
         unit_count = line_item.inventory_units.size
         return if unit_count >= line_item.quantity
         quantity = line_item.quantity - unit_count
+        return if quantity.zero?
 
         quantifier = Stock::Quantifier.new(line_item.variant)
 
-        unless quantifier.can_supply? quantity
-          variant = line_item.variant
-          display_name = %Q{#{variant.name}}
-          display_name += %Q{ (#{variant.options_text})} unless variant.options_text.blank?
+        return if quantifier.can_supply?(quantity)
 
-          line_item.errors[:quantity] << Spree.t(
-            :selected_quantity_not_available,
-            item: display_name.inspect
-          )
-        end
+        variant = line_item.variant
+        display_name = %Q{#{variant.name}}
```

Comment:
Use `%` instead of `%Q`.<br>Use `%Q` only for strings that contain both single quotes and double quotes, or for dynamic strings that contain double quotes.